aten/src/ATen/native/nested/NestedTensorMath.cpp

```diff
@@ -688,17 +688,35 @@ Tensor select_nested(const Tensor& self, int64_t dim, int64_t index) {
 Tensor clone_nested(
     const Tensor& self,
     c10::optional<c10::MemoryFormat> optional_memory_format) {
-  auto memory_format = optional_memory_format.value_or(MemoryFormat::Preserve);
-  TORCH_CHECK(
-      memory_format == MemoryFormat::Preserve,
-      "clone_nested only supports memory format Preserve, but got ",
-      memory_format,
-      " instead.");
-  // TODO: The size doesn't necessarily need to be cloned, but it is more
-  // conservative. This is something we could revisit once we land a more
-  // efficient implementation of nested_size_tensor_.
-  return wrap_buffer(
-      get_buffer(self).clone(), get_nested_size_tensor(self).clone());
+  auto memory_format = optional_memory_format.value_or(c10::MemoryFormat::Preserve);
+  auto self_ptr = get_nested_tensor_impl(self);
+  if (memory_format == c10::MemoryFormat::Preserve ||
+  (memory_format == c10::MemoryFormat::Contiguous && nested_tensor_impl_is_contiguous(self_ptr))) {
+    Tensor buffer = self_ptr->get_buffer().clone();
+    const Tensor& sizemat = self_ptr->get_nested_size_tensor(),
+        stridemat = self_ptr->get_nested_stride_tensor();
+    const std::vector<int64_t>& offsets = self_ptr->get_offsets();
+    // TODO: The size and the stride do not necessarily need to be cloned,
+    //       but it is more conservative.
+    //       This is something we could revisit once we land a more
+    //       efficient implementation of nested_size_tensor_ and nested_stride_tensor.
+    return wrap_buffer(buffer, sizemat.clone(), stridemat.clone(), offsets);
+  }
+  else if (memory_format == c10::MemoryFormat::Contiguous) {
```

Comment:
I would maybe add a comment that is the case when the memory format is contiguous but self is a non_contiguous nested tensor. 

One thing I am trying to hash out now. I think all current nested tensor instances satisfy the bottom two are equal.

`nested_tensor.numel()` -> this is calculated from nested_sizes to have less elements than its underlying 
storage ->` nested_tensor.storage.nbytes()//dtype`. 

In theory one can easily create this tensor with wrap_buffer(), however I am not sure if that is done anywhere. 

reshape, transpose, view / all current view ops appear to maintain this invariant but need to confirm.
@jbschlosser

Reply:
slice is an example to have `nested_tensor.numel()` < `nested_tensor.storage.nbytes()//dtype`
```
nt = torch.nested_tensor([x, y, z])
slice = nt[0 : 2 : 2]
```
where the memory of `y` is skipped